rclcpp_lifecycle/src/transition.cpp

```diff
@@ -99,4 +156,47 @@ Transition::goal_state() const
   return State(transition_handle_->goal);
 }
 
+void
+Transition::reset()
+{
+  // nothing to free here
+  if (!transition_handle_) {
+    owns_rcl_transition_handle_ = false;
+    return;
+  }
+
+  // can't free anything which is not owned
+  if (!owns_rcl_transition_handle_) {
+    transition_handle_ = nullptr;
+    return;
+  }
+
+  if (transition_handle_->start) {
+    if (transition_handle_->start->label) {
+      allocator_.deallocate(
+        const_cast<char *>(transition_handle_->start->label), allocator_.state);
```

Comment:
The need for a `const_cast` here indicates that the field `transition_handle_->start` / `transition_handle_->start->label` should probably not be declared as `const`. But changing that might be out of scope of this PR.